cupy/random/cupy_distributions.cu

```diff
@@ -362,14 +364,43 @@ struct standard_normal_float_functor {
     }
 };
 
+// There are several errors when trying to do this a full template
+struct standard_gamma_functor {
+    template<typename... Args>
+    __device__ double operator () (Args&&... args) {
+        return rk_standard_gamma(args...);
+    }
+};
+
+// The following templates are used to unwrap arrays into an elementwise
+// approach
```

Comment:
Could you mention that the wrapper is `_array_data` in `cupy/random/_generator_api.pyx`?